### PR TITLE
fix: dead/powered down barricade is not predicted

### PR DESCRIPTION
**Pull request by cu-kai**
take 2

### CODE REVIEW

File: src/cgame/cg_predict.cpp [hunk 1]
```diff
@@ -147,7 +147,7 @@ static void CG_ClipMoveToEntities( const vec3_t start, const vec3_t mins,
 			if ( ent->eType == entityType_t::ET_BUILDABLE )
 			{
 				BG_BuildableBoundingBox( ent->modelindex, bmins, bmaxs );
-				if ( ent->modelindex == BA_A_BARRICADE && ent->torsoAnim == BANIM_IDLE_UNPOWERED )
+				if ( ent->modelindex == BA_A_BARRICADE && ( ent->torsoAnim != BANIM_IDLE1 || !( ent->eFlags & EF_B_SPAWNED ) ) )
```
Comment (DolceTriade) [inline src/cgame/cg_predict.cpp:150]: Looking closer at my original fix, I wonder if we could just do `ent->legsAnim == BANIM_POWERDOWN` as the condition and catch everything...

Comment (DolceTriade) [reply src/cgame/cg_predict.cpp:150]: Seems to work ok in my tests?

**Comment by DolceTriade** [comment]
Ok, before doing some extensive redesign, I think it's fair to say this improves the situation. Perhaps it is fine to submit as is to improve the situation and then we can implement the full design after?

**Comment by slipher** [comment]
I'm not demanding the use of any specific approach. But if anyone is interested in trying the old version with bboxes calculated by the server, they can revert 7f00ec84a911c8b0607025aeefe53be60b95f3d9.

File: src/cgame/cg_predict.cpp (hunk 1 above)
Comment (DolceTriade) [reply src/cgame/cg_predict.cpp:150]: eh, it's meh. I think add a note here saying `// TODO: This is not perfect and can be improved.`

**Comment by DolceTriade** [comment]
LGTM

**Pull request merged**